In [14]:
from sklearn import datasets, metrics
from sklearn.ensemble import GradientBoostingClassifier,RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier, DecisionTreeRegressor
from sklearn.model_selection import train_test_split

# 作業  
目前已經學過許多的模型，相信大家對整體流程應該比較掌握了，這次作業請改用**手寫辨識資料集**，步驟流程都是一樣的，請試著自己撰寫程式碼來完成所有步驟

In [4]:
digits = datasets.load_digits()

### GBC

In [12]:
# 切分訓練/測試集
x_train, x_test, y_train, y_test = train_test_split(digits.data, digits.target, test_size=0.25, random_state=4)

# 建GBC模
clf = GradientBoostingClassifier()

# 訓練模型
clf.fit(x_train, y_train)

# 預測測試集
y_pred = clf.predict(x_test)

acc = metrics.accuracy_score(y_test, y_pred)
print("Acuuracy: ", acc)

Acuuracy:  0.9644444444444444


### Random Forest

In [16]:
# 建立隨機森林模型
RF = RandomForestClassifier()

# 訓練模型
RF.fit(x_train, y_train)

# 預測測試集
RF_y_pred = RF.predict(x_test)

acc_rf = metrics.accuracy_score(y_test,RF_y_pred)
print("Acuuracy: ", acc_rf)

Acuuracy:  0.9533333333333334


### Decision Tree

In [17]:
# 建立決策樹模型
DTC = DecisionTreeClassifier()

# 將訓練資料丟進去模型訓練
DTC.fit(x_train, y_train)

# 將測試資料丟進模型得到預測結果
DTC_y_pred = DTC.predict(x_test)

acc_dt = metrics.accuracy_score(y_test,DTC_y_pred)
print("Acuuracy: ", acc_dt)

Acuuracy:  0.8644444444444445


→  
GBC表現最好，Acuuracy=0.964；  
Random Forest次之，Acuuracy=0.953；  
決策樹最差，Acuuracy=0.864  

# 範例

In [2]:
# 讀取鳶尾花資料集
iris = datasets.load_iris()

# 切分訓練集/測試集
x_train, x_test, y_train, y_test = train_test_split(iris.data, iris.target, test_size=0.25, random_state=4)

# 建立模型
clf = GradientBoostingClassifier()

# 訓練模型
clf.fit(x_train, y_train)

# 預測測試集
y_pred = clf.predict(x_test)

In [3]:
acc = metrics.accuracy_score(y_test, y_pred)
print("Acuuracy: ", acc)

Acuuracy:  0.9736842105263158
